library/azure_rm_rediscache.py

```diff
@@ -0,0 +1,618 @@
+            host_name=None
+        )
+        self.state = None
+        self.to_do = Actions.NoAction
+
+        self.frameworks = None
+
+        super(AzureRMRedisCaches, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                             supports_check_mode=True,
+                                             supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        old_response = None
+        response = None
+        to_be_updated = False
+
+        # define redis_configuration properties
+        self.redis_configuration_properties = ["maxfragmentationmemory_reserved",
+                                               "maxmemory_reserved",
+                                               "maxmemory_policy",
+                                               "notify_keyspace_events"]
+
+        # get management client
+        self._client = self.get_mgmt_svc_client(RedisManagementClient,
+                                                base_url=self._cloud_environment.endpoints.resource_manager,
+                                                api_version='2018-03-01')
+
+        # set location
+        resource_group = self.get_resource_group(self.resource_group)
+        if not self.location:
+            self.location = resource_group.location
+
+        # check subnet exists
+        if self.subnet:
+            self.subnet = self.parse_subnet()
+
+        #ubnet = self.parse_subnet()get existing redis cache
+        old_response = self.get_rediscache()
+
+        if old_response:
+            self.results['id'] = old_response['id']
+
+        if self.state == 'present':
+            # if redis not exists
+            if not old_response:
+                self.log("Redis cache instance doesn't exist")
+
+                to_be_updated = True
+                self.to_do = Actions.Create
+                
+                if not self.sku:
+                    self.fail("Please specify sku to creating new redis cache.")
+
+            else:
+                # redis exists already, do update
+                self.log("Redis cache instance already exists")
+
+                update_tags, self.tags = self.update_tags(old_response.get('tags', None))
+
+                if update_tags:
+                    to_be_updated = True
+                    self.to_do = Actions.Update
+
+                # check if update
+                if self.check_update(old_response):
+                    to_be_updated = True
+                    self.to_do = Actions.Update
+
+        elif self.state == 'absent':
+            if old_response:
+                self.log("Delete Redis cache instance")
+                self.results['changed'] = True
+
+                if self.check_mode:
+                    return self.results
+
+                self.delete_rediscache()
```

Comment:
Why not keep the consistent to assign an Action.Delete and handle the action in the action execute part?

Reply:
fixed